src/main/java/org/wildfly/security/auth/server/IdentityCredentials.java

```diff
@@ -294,6 +296,25 @@ public IdentityCredentials without(final Class<? extends Credential> credentialT
         return Spliterators.spliterator(iterator(), size(), Spliterator.IMMUTABLE | Spliterator.DISTINCT | Spliterator.NONNULL | Spliterator.ORDERED | Spliterator.SIZED);
     }
 
+    public boolean canVerify(Class<? extends Evidence> evidenceClass, String algorithmName) {
+        return StreamSupport.stream(spliterator(), false)
+                .filter(credential -> credential.canVerify(evidenceClass, algorithmName))
+                .count() != 0;
```

Comment:
A small nit, but maybe ``findAny().isPresent()`` is a better choice here?  Only fix this if you have to do another iteration of changes though, otherwise don't worry about it.